extensions/subtis/src/index.tsx

```diff
@@ -0,0 +1,350 @@
+import {
+  Action,
+  ActionPanel,
+  Form,
+  Toast,
+  open,
+  showToast,
+} from "@raycast/api";
+import { stat } from "fs/promises";
+import fetch from "node-fetch";
+import { match } from "ts-pattern";
+import WebSocket from "ws";
+import { z } from "zod";
+
+// constants
+// const API_URL =
+//   process.env.NODE_ENV === "production"
+//     ? "https://api.subt.is"
+//     : "http://localhost:58602";
+const API_URL = "https://api.subt.is";
+const VIDEO_FILE_EXTENSIONS = [
+  ".mkv",
+  ".mp4",
+  ".avi",
+  ".mov",
+  ".wmv",
+  ".flv",
+  ".webm",
+  ".vob",
+  ".m4v",
+  ".mpg",
+  ".mpeg",
+  ".3gp",
+  ".3g2",
+] as const;
+
+// helpers
+function getFilenameFromPath(path: string): string {
+  return path.split(/[\\/]/).at(-1) as string;
+}
+
+function getMessageFromStatusCode(statusCode: number): {
+  description: string;
+  title: string;
+} {
+  return match(statusCode)
+    .with(415, () => ({
+      title: "Extensión de archivo no soportada",
+      description: "Prueba con formatos como MKV, MP4, o AVI",
+    }))
+    .with(404, () => ({
+      title: "Subtitulo no encontrado",
+      description: "Nos pondremos a buscarlo",
+    }))
+    .with(500, () => ({
+      title: "Error Inesperado",
+      description: "Estamos haciendo arreglos del servicio",
+    }))
+    .otherwise(() => ({
+      title: "Error desconocido",
+      description: "Estamos haciendo arreglos del servicio",
+    }));
+}
```

Comment:
At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension. If the locale might affect functionality (e.g. using the correct unit of measurement), please use the preferences API.